File: android/app/src/main/java/com/freelancekar/app/ui/screens/ProfileScreen.kt
package com.freelancekar.app.ui.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.unit.dp
import com.freelancekar.app.data.*
import com.freelancekar.app.ui.AppViewModel
import com.freelancekar.app.ui.components.*
import com.freelancekar.app.ui.theme.FkColor
import com.freelancekar.app.ui.theme.FkType

@Composable
fun ProfileScreen(state: AppState, vm: AppViewModel) {
    val pro = activePro(state)
    val saved = state.saved.contains(pro.id)

    Column(Modifier.fillMaxSize().verticalScroll(rememberScrollState())) {
        Box(Modifier.fillMaxWidth()) {
            Stripe(Modifier.fillMaxWidth().padding(top = 12.dp).height(104.dp))
            Box(
                Modifier
                    .padding(top = 12.dp, start = 16.dp)
                    .size(32.dp)
                    .clip(CircleShape)
                    .background(FkColor.bg.copy(alpha = 0.72f))
                    .border(1.dp, FkColor.line12, CircleShape)
                    .clickable { vm.go(if (state.searchLabel.isNotEmpty()) Screen.RESULTS else Screen.HOME) },
                contentAlignment = Alignment.Center,
            ) {
                Text("←", style = FkType.body.copy(color = FkColor.text))
            }
        }

        Row(
            Modifier.fillMaxWidth().offset(y = (-30).dp).padding(horizontal = 20.dp),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.Bottom,
        ) {
            Stripe(Modifier.size(72.dp).clip(RoundedCornerShape(20.dp)).border(3.dp, FkColor.bg, RoundedCornerShape(20.dp)))
            Row(horizontalArrangement = Arrangement.spacedBy(8.dp), modifier = Modifier.padding(bottom = 4.dp)) {
                Box(
                    Modifier
                        .clip(RoundedCornerShape(10.dp))
                        .border(1.dp, FkColor.line14, RoundedCornerShape(10.dp))
                        .clickable { vm.toggleSave(pro.id, pro.first) }
                        .padding(horizontal = 11.dp, vertical = 7.dp),
                ) {
                    Text(if (saved) "♥" else "♡", style = FkType.body.copy(color = if (saved) FkColor.lime else FkColor.text.copy(alpha = 0.7f)))
                }
                Box(
                    Modifier
                        .clip(RoundedCornerShape(10.dp))
                        .border(1.dp, FkColor.line14, RoundedCornerShape(10.dp))
                        .padding(horizontal = 11.dp, vertical = 7.dp),
                ) {
                    Text("↗", style = FkType.body.copy(color = FkColor.text.copy(alpha = 0.7f)))
                }
            }
        }

        Column(Modifier.padding(horizontal = 20.dp).offset(y = (-18).dp)) {
            Row(verticalAlignment = Alignment.CenterVertically) {
                Text(pro.name, style = FkType.h1sm)
                Spacer(Modifier.width(7.dp))
                Box(
                    Modifier.clip(RoundedCornerShape(6.dp)).background(FkColor.limeBadge).padding(horizontal = 7.dp, vertical = 3.dp),
                ) {
                    Text("KYC ✔", style = FkType.monoSm.copy(color = FkColor.lime))
                }
            }
            Spacer(Modifier.height(6.dp))
            Text(pro.tagline, style = FkType.body.copy(color = FkColor.text55))
        }

        Row(
            Modifier.padding(horizontal = 20.dp).padding(top = 16.dp),
            horizontalArrangement = Arrangement.spacedBy(8.dp),
        ) {
            StatBox(pro.rating.toString(), "${pro.jobs} jobs", FkColor.amber, Modifier.weight(1f))
            StatBox("98%", "completion", null, Modifier.weight(1f))
            StatBox(pro.replyIn, "replies in", null, Modifier.weight(1f))
            StatBox(pro.years, "on platform", null, Modifier.weight(1f))
        }

        Column(Modifier.padding(horizontal = 20.dp).padding(top = 24.dp)) {
            SectionTitle(
                "Services offered",
                right = { Text("${pro.packages.size} ACTIVE", style = FkType.mono.copy(color = FkColor.text40)) },
            )
            Spacer(Modifier.height(11.dp))
            Column(verticalArrangement = Arrangement.spacedBy(9.dp)) {
                pro.packages.forEachIndexed { i, p ->
                    Row(
                        Modifier
                            .fillMaxWidth()
                            .clip(RoundedCornerShape(14.dp))
                            .background(FkColor.card)
                            .border(1.dp, if (i == state.svcIdx) FkColor.lime else FkColor.line08, RoundedCornerShape(14.dp))
                            .clickable { vm.pickService(i) }
                            .padding(horizontal = 14.dp, vertical = 13.dp),
                        horizontalArrangement = Arrangement.SpaceBetween,
                        verticalAlignment = Alignment.CenterVertically,
                    ) {
                        Column(verticalArrangement = Arrangement.spacedBy(3.dp)) {
                            Text(p.name, style = FkType.cardTitle)
                            Text(p.terms, style = FkType.meta.copy(color = FkColor.text50))
                        }
                        Text(p.price, style = FkType.body.copy(fontWeight = androidx.compose.ui.text.font.FontWeight.SemiBold))
                    }
                }
            }
        }

        Column(Modifier.padding(horizontal = 20.dp).padding(top = 24.dp)) {
            Text("Recent work", style = FkType.h3)
            Spacer(Modifier.height(11.dp))
            Row(horizontalArrangement = Arrangement.spacedBy(7.dp)) {
                Stripe(Modifier.weight(1f).aspectRatio(3f / 4f).clip(RoundedCornerShape(11.dp)))
                Stripe(Modifier.weight(1f).aspectRatio(3f / 4f).clip(RoundedCornerShape(11.dp)))
                Stripe(Modifier.weight(1f).aspectRatio(3f / 4f).clip(RoundedCornerShape(11.dp))) {
                    Text("+14", style = FkType.h4.copy(color = FkColor.text60))
                }
            }
        }

        Column(Modifier.padding(horizontal = 20.dp).padding(top = 24.dp, bottom = 30.dp)) {
            Text("Reviews", style = FkType.h3)
            Spacer(Modifier.height(11.dp))
            Column(
                Modifier
                    .fillMaxWidth()
                    .clip(RoundedCornerShape(14.dp))
                    .background(FkColor.card)
                    .border(1.dp, FkColor.line08, RoundedCornerShape(14.dp))
                    .padding(14.dp),
            ) {
                Row(verticalAlignment = Alignment.CenterVertically) {
                    Stripe(Modifier.size(28.dp).clip(CircleShape))
                    Spacer(Modifier.width(9.dp))
                    Text(pro.reviewer, style = FkType.body.copy(fontWeight = androidx.compose.ui.text.font.FontWeight.SemiBold, fontSize = androidx.compose.ui.unit.TextUnit(12.5f, androidx.compose.ui.unit.TextUnitType.Sp)))
                    Spacer(Modifier.weight(1f))
                    Text("★ 5.0", style = FkType.meta.copy(color = FkColor.amber))
                }
                Spacer(Modifier.height(10.dp))
                Text(pro.review, style = FkType.bodySm.copy(color = FkColor.text60))
            }
        }
    }
}

@Composable
private fun StatBox(value: String, label: String, valueColor: androidx.compose.ui.graphics.Color?, modifier: Modifier = Modifier) {
    Column(
        modifier
            .clip(RoundedCornerShape(12.dp))
            .background(FkColor.card)
            .border(1.dp, FkColor.line08, RoundedCornerShape(12.dp))
            .padding(horizontal = 4.dp, vertical = 10.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
    ) {
        Text(value, style = FkType.body.copy(fontWeight = androidx.compose.ui.text.font.FontWeight.SemiBold, fontSize = androidx.compose.ui.unit.TextUnit(14f, androidx.compose.ui.unit.TextUnitType.Sp), color = valueColor ?: FkColor.text))
        Spacer(Modifier.height(2.dp))
        Text(label, style = FkType.metaXs.copy(color = FkColor.text45))
    }
}

/** Replaces the tab bar while a freelancer profile is open. */
@Composable
fun ProfileActionBar(state: AppState, vm: AppViewModel) {
    val pro = activePro(state)
    val pkg = selectedPackage(state)
    Row(
        Modifier
            .fillMaxWidth()
            .background(FkColor.bar)
            .border(1.dp, FkColor.line07)
            .padding(horizontal = 20.dp)
            .padding(top = 14.dp, bottom = 20.dp),
        horizontalArrangement = Arrangement.SpaceBetween,
        verticalAlignment = Alignment.CenterVertically,
    ) {
        Column {
            Text(pkg.price, style = FkType.body.copy(fontWeight = androidx.compose.ui.text.font.FontWeight.SemiBold, fontSize = androidx.compose.ui.unit.TextUnit(15f, androidx.compose.ui.unit.TextUnitType.Sp)))
            Text(pkg.terms, style = FkType.metaXs.copy(color = FkColor.text45))
        }
        Row(horizontalArrangement = Arrangement.spacedBy(8.dp)) {
            InlineOutlineButton({ vm.messagePro(pro) }, text = "Chat")
            InlineButton({ vm.openHire() }, text = "Hire")
        }
    }
}

File: ios/FreelanceKar/Views/Screens/ProfileScreen.swift
import SwiftUI

/// Client-facing view of a freelancer profile. Mirrors Profile.tsx.
struct ProfileScreen: View {
    @EnvironmentObject var store: AppStore

    var body: some View {
        let s = store.state
        let pro = Selectors.activePro(s)
        let saved = s.saved.contains(pro.id)

        ScreenShell {
            VStack(alignment: .leading, spacing: 0) {
                ZStack(alignment: .topLeading) {
                    Stripe(cornerRadius: 0).frame(height: 104).padding(.top, 12)
                    Button {
                        store.go(s.searchLabel.isEmpty ? .home : .results)
                    } label: {
                        Image(systemName: "chevron.left").font(.system(size: 14)).foregroundStyle(FK.text)
                            .frame(width: 32, height: 32)
                            .background(Color(hex: 0x0B0B0C, opacity: 0.72))
                            .clipShape(Circle())
                            .overlay(Circle().stroke(FK.line12, lineWidth: 1))
                    }
                    .padding(.leading, 16).padding(.top, 24)
                }

                HStack(alignment: .bottom) {
                    Stripe(cornerRadius: 20).frame(width: 72, height: 72)
                        .overlay(RoundedRectangle(cornerRadius: 20).stroke(FK.bg, lineWidth: 3))
                    Spacer()
                    HStack(spacing: 8) {
                        Button {
                            store.toggleSave(pro.id, pro.first)
                        } label: {
                            Text(saved ? "♥" : "♡").font(.system(size: 13))
                                .foregroundStyle(saved ? FK.lime : Color(hex: 0xF2F2F0, opacity: 0.7))
                                .padding(.horizontal, 11).padding(.vertical, 7)
                                .overlay(RoundedRectangle(cornerRadius: 10).stroke(FK.line14, lineWidth: 1))
                        }
                        Text("↗").font(.system(size: 13)).foregroundStyle(Color(hex: 0xF2F2F0, opacity: 0.7))
                            .padding(.horizontal, 11).padding(.vertical, 7)
                            .overlay(RoundedRectangle(cornerRadius: 10).stroke(FK.line14, lineWidth: 1))
                    }
                }
                .padding(.horizontal, 20)
                .offset(y: -30)
                .padding(.bottom, -30)

                VStack(alignment: .leading, spacing: 6) {
                    HStack(spacing: 7) {
                        Text(pro.name).font(.fkH1Sm)
                        Text("KYC ✔").font(.system(size: 10, weight: .semibold, design: .monospaced))
                            .foregroundStyle(FK.lime)
                            .padding(.horizontal, 7).padding(.vertical, 3)
                            .background(FK.limeBadge)
                            .clipShape(RoundedRectangle(cornerRadius: 6, style: .continuous))
                    }
                    Text(pro.tagline).font(.system(size: 13)).foregroundStyle(FK.text55)
                }
                .padding(.horizontal, 20)
                .padding(.top, 12)

                HStack(spacing: 8) {
                    ProfileStatBox(value: String(format: "%.1f", pro.rating), label: "\(pro.jobs) jobs", valueColor: FK.amber)
                    ProfileStatBox(value: "98%", label: "completion")
                    ProfileStatBox(value: pro.replyIn, label: "replies in")
                    ProfileStatBox(value: pro.years, label: "on platform")
                }
                .padding(.horizontal, 20).padding(.top, 16)

                VStack(alignment: .leading, spacing: 11) {
                    SectionTitle(title: "Services offered") {
                        Text("\(pro.packages.count) ACTIVE").font(.system(size: 11, design: .monospaced)).foregroundStyle(FK.text40)
                    }
                    ForEach(Array(pro.packages.enumerated()), id: \.element.id) { i, p in
                        Button { store.pickService(i) } label: {
                            HStack {
                                VStack(alignment: .leading, spacing: 3) {
                                    Text(p.name).font(.fkCardTitle)
                                    Text(p.terms).font(.system(size: 11)).foregroundStyle(FK.text50)
                                }
                                Spacer()
                                Text(p.price).font(.system(size: 13, weight: .semibold))
                            }
                            .padding(.horizontal, 14).padding(.vertical, 13)
                        }
                        .foregroundStyle(FK.text)
                        .background(FK.card)
                        .clipShape(RoundedRectangle(cornerRadius: 14, style: .continuous))
                        .overlay(RoundedRectangle(cornerRadius: 14).stroke(i == s.svcIdx ? FK.lime : FK.line08, lineWidth: 1))
                    }
                }
                .padding(.horizontal, 20).padding(.top, 24)

                VStack(alignment: .leading, spacing: 11) {
                    Text("Recent work").font(.fkH3)
                    HStack(spacing: 7) {
                        Stripe(cornerRadius: 11).aspectRatio(3.0/4.0, contentMode: .fit)
                        Stripe(cornerRadius: 11).aspectRatio(3.0/4.0, contentMode: .fit)
                        Stripe(cornerRadius: 11).aspectRatio(3.0/4.0, contentMode: .fit)
                            .overlay(Text("+14").font(.system(size: 12, weight: .medium)).foregroundStyle(FK.text60))
                    }
                }
                .padding(.horizontal, 20).padding(.top, 24)

                VStack(alignment: .leading, spacing: 11) {
                    Text("Reviews").font(.fkH3)
                    VStack(alignment: .leading, spacing: 10) {
                        HStack(spacing: 9) {
                            Circle().fill(FK.nested).frame(width: 28, height: 28)
                            Text(pro.reviewer).font(.system(size: 12.5, weight: .semibold))
                            Spacer()
                            Text("★ 5.0").font(.system(size: 11, weight: .medium)).foregroundStyle(FK.amber)
                        }
                        Text(pro.review).font(.fkBodySm).foregroundStyle(FK.text60)
                    }
                    .padding(14)
                    .background(FK.card)
                    .clipShape(RoundedRectangle(cornerRadius: 14, style: .continuous))
                    .overlay(RoundedRectangle(cornerRadius: 14).stroke(FK.line08, lineWidth: 1))
                }
                .padding(.horizontal, 20).padding(.top, 24).padding(.bottom, 30)
            }
        }
    }
}

private struct ProfileStatBox: View {
    var value: String
    var label: String
    var valueColor: Color = FK.text
    var body: some View {
        VStack(spacing: 2) {
            Text(value).font(.system(size: 14, weight: .semibold)).foregroundStyle(valueColor)
            Text(label).font(.system(size: 9.5)).foregroundStyle(FK.text45)
        }
        .frame(maxWidth: .infinity)
        .padding(.vertical, 10)
        .background(FK.card)
        .clipShape(RoundedRectangle(cornerRadius: 12, style: .continuous))
        .overlay(RoundedRectangle(cornerRadius: 12).stroke(FK.line08, lineWidth: 1))
    }
}

/// Replaces the tab bar while a freelancer profile is open.
struct ProfileActionBar: View {
    @EnvironmentObject var store: AppStore
    var body: some View {
        let s = store.state
        let pro = Selectors.activePro(s)
        let pkg = Selectors.selectedPackage(s)

        HStack {
            VStack(alignment: .leading, spacing: 0) {
                Text(pkg.price).font(.system(size: 15, weight: .semibold))
                Text(pkg.terms).font(.system(size: 10.5)).foregroundStyle(FK.text45)
            }
            Spacer()
            HStack(spacing: 8) {
                InlineOutlineButton(title: "Chat") { store.messagePro(pro) }
                InlineButton(title: "Hire") { store.openHire() }
            }
        }
        .padding(.horizontal, 20).padding(.top, 14).padding(.bottom, 20)
        .background(FK.bar)
        .overlay(Rectangle().frame(height: 1).foregroundStyle(FK.line07), alignment: .top)
    }
}
